src/osd/OSD.cc

```diff
@@ -9938,9 +9946,45 @@ void OSD::handle_conf_change(const ConfigProxy& conf,
 			     const std::set <std::string> &changed)
 {
   std::lock_guard l{osd_lock};
-  if (changed.count("osd_max_backfills")) {
-    service.local_reserver.set_max(cct->_conf->osd_max_backfills);
-    service.remote_reserver.set_max(cct->_conf->osd_max_backfills);
+
+  if (changed.count("osd_max_backfills") ||
+      changed.count("osd_recovery_sleep") ||
+      changed.count("osd_recovery_sleep_hdd") ||
+      changed.count("osd_recovery_sleep_ssd") ||
+      changed.count("osd_recovery_sleep_hybrid") ||
+      changed.count("osd_recovery_max_active") ||
+      changed.count("osd_recovery_max_active_hdd") ||
+      changed.count("osd_recovery_max_active_ssd") ||
+      changed.count("osd_async_recovery_min_cost")) {
+    if (cct->_conf.get_val<std::string>("osd_op_queue") == "mclock_scheduler" &&
+        cct->_conf.get_val<std::string>("osd_mclock_profile") != "custom") {
+      // Set ceph config option to meet QoS goals
+      // Set low recovery min cost
+      cct->_conf.set_val("osd_async_recovery_min_cost", stringify(1));
+      // Set high value for recovery max active
+      uint32_t recovery_max_active = 1000;
+      if (cct->_conf->osd_recovery_max_active)
+        cct->_conf.set_val(
+            "osd_recovery_max_active", stringify(recovery_max_active));
+      if (store_is_rotational)
+        cct->_conf.set_val(
+            "osd_recovery_max_active_hdd", stringify(recovery_max_active));
+      else
+        cct->_conf.set_val(
+            "osd_recovery_max_active_ssd",stringify(recovery_max_active));
+      // Set high value for osd_max_backfill
+      cct->_conf.set_val("osd_max_backfills", stringify(1000));
+
+      // Disable recovery sleep
+      cct->_conf.set_val("osd_recovery_sleep", stringify(0));
+      cct->_conf.set_val("osd_recovery_sleep_hdd", stringify(0));
+      cct->_conf.set_val("osd_recovery_sleep_ssd", stringify(0));
+      cct->_conf.set_val("osd_recovery_sleep_hybrid", stringify(0));
+    }
```

Comment:
nit: else should be on same line as braces

Reply:
done